sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -85,6 +85,157 @@ def test_mean_variance_axis1():
+                                                     last_var=last_var,
+                                                     last_n=last_n,
+                                                     weights=None)
+
+    means_w0, vars_w0, n_incr_w0 = incr_mean_variance_axis(
+            X=Xw_sparse, axis=axis, last_mean=last_mean, last_var=last_var,
+            last_n=last_n, weights=weights)
+
+    means_simple, vars_simple = mean_variance_axis(X=X_sparse, axis=axis)
+
+    assert_array_almost_equal(means0, means_w0)
+    assert_array_almost_equal(means0, means_simple)
+    assert_array_almost_equal(vars0, vars_w0)
+    assert_array_almost_equal(vars0, vars_simple)
+    assert_array_almost_equal(n_incr0, n_incr_w0)
+
+    # check second round for incremental
+    means1, vars1, n_incr1 = incr_mean_variance_axis(X=X_sparse, axis=axis,
+                                                     last_mean=means0,
+                                                     last_var=vars0,
+                                                     last_n=n_incr0,
+                                                     weights=None)
+
+    means_w1, vars_w1, n_incr_w1 = incr_mean_variance_axis(
+            X=Xw_sparse, axis=axis, last_mean=means_w0, last_var=vars_w0,
+            last_n=n_incr_w0, weights=weights)
+
+    assert_array_almost_equal(means1, means_w1)
+    assert_array_almost_equal(vars1, vars_w1)
+    assert_array_almost_equal(n_incr1, n_incr_w1)
+
+
+@pytest.mark.parametrize(['Xw', 'X', 'weights'],
+                         [
+                         ([[0, 0, 1], [0, 2, 3]],
+                          [[0, 0, 1], [0, 2, 3]],
+                          [1, 1]),
+                         ([[0, 0, 1], [0, 1, 1]],
+                          [[0, 0, 1], [0, 1, 1], [0, 1, 1]],
+                          [1, 2]),
+                         ([[0, 0, 1], [0, 1, 1]],
+                          [[0, 0, 1], [0, 1, 1]],
+                          None),
+                         ([[0, np.nan, 2],
+                           [0, np.nan, np.nan]],
+                          [[0, np.nan, 2],
+                           [0, np.nan, np.nan]],
+                          [1., 1.]),
+                         ([[0, 0, 1, np.nan, 2, 0],
+                           [0, 3, np.nan, np.nan, np.nan, 2]],
+                          [[0, 0, 1, np.nan, 2, 0],
+                           [0, 0, 1, np.nan, 2, 0],
+                           [0, 3, np.nan, np.nan, np.nan, 2]],
+                          [2., 1.]),
+                         ([[1, 0, 1], [0, 0, 1]],
+                          [[1, 0, 1], [0, 0, 1], [0, 0, 1], [0, 0, 1]],
+                          np.array([1, 3]))
+                         ]
+                         )
+@pytest.mark.parametrize("sparse_constructor",
+                         [sp.csc_matrix, sp.csr_matrix])
+def test_incr_mean_variance_axis_weighted_axis0(Xw, X, weights,
+                                                sparse_constructor):
```

Comment:
Could you please also parametrize with `dtype` in `(np.float64, np.float32)` and check that the dtype of the returned arrays match the input dtype?